### PR TITLE
Vendor listing code with LLD

### PR DESCRIPTION
**Pull request by NevinJoseph22**
Screenshot 2024-08-21 155646

### CODE REVIEW

File: src/app/features/vendor/models/vendor.model.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface Vendor {
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/models/vendor.model.ts]: Interface name should start with 'I'(IVendor here)

File: src/app/features/vendor/vendor-listing/listing.component.html [hunk 2]
```diff
@@ -1 +1,35 @@
...
+    <p-table
+      [value]="vendors"
+      [paginator]="true"
+      [rows]="5"
+      [tableStyle]="{ 'min-width': '50rem' }"
+      [rowsPerPageOptions]="[5, 10, 20]"
+    >
+      <ng-template pTemplate="header">
+        <tr>
+          <th style="width: 25%">Vendor Name</th>
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-listing/listing.component.html]: Remove inline styling

File: src/app/features/vendor/models/vendor.model.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  id: string;
+  name: string;
+  state?:string;
+  country:string;
+  markets: string[];
```
Comment (kavyamolr) [inline src/app/features/vendor/models/vendor.model.ts:6]: File is not formatted. (space is missing after colon)

File: src/app/features/vendor/vendor-listing/listing.component.html [hunk 4]
```diff
@@ -1 +1,35 @@
-<p>listing works!</p>
+<div class="page-container">
+  <div class="header-container">
+    <h2>All Vendors</h2>
+    <button class="add-vendor-button" [routerLink]="['/vendor/creation']">+ Add New Vendor</button>
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/listing.component.html]: add type and id for button.
id should be unique. eg: 'add-new-vendor-btn'

File: src/app/features/vendor/vendor-listing/listing.component.ts [hunk 5]
```diff
@@ -1,10 +1,99 @@
-import { Component } from '@angular/core';
-//import { CommonModule } from '@angular/common';
+import { Component,OnInit } from '@angular/core';
+import { Vendor } from '../models/vendor.model';
+
+// LLD
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/listing.component.ts]: Please refer the sample LLD and follow the same pattern

File: src/app/features/vendor/vendor-listing/listing.component.ts [hunk 6]
```diff
@@ -1,10 +1,99 @@
...
 
 @Component({
   selector: 'app-listing',
   standalone: false,
   templateUrl: './listing.component.html',
   styleUrl: './listing.component.scss',
 })
-export class ListingComponent {}
+export class ListingComponent implements OnInit {
+  vendors!: Vendor[];
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/listing.component.ts]: Add an empty line between methods.

File: src/app/features/vendor/models/vendor.model.ts (hunk 1 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/models/vendor.model.ts]: updated

File: src/app/features/vendor/vendor-listing/listing.component.html (hunk 2 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/listing.component.html]: removed

File: src/app/features/vendor/models/vendor.model.ts (hunk 3 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/models/vendor.model.ts:6]: format corrected

File: src/app/features/vendor/vendor-listing/listing.component.html (hunk 4 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/listing.component.html]: added

File: src/app/features/vendor/vendor-listing/listing.component.ts (hunk 5 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/listing.component.ts]: LLD updated

File: src/app/features/vendor/vendor-listing/listing.component.ts (hunk 6 above)
Comment (NevinJoseph22) [reply src/app/features/vendor/vendor-listing/listing.component.ts]: Added

File: src/app/features/vendor/vendor-listing/listing.component.html [hunk 7]
```diff
@@ -1 +1,37 @@
-<p>listing works!</p>
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/listing.component.html:1]: Provide appropriate component name

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+ * Default sort order: Ascending order of vendor names.
+ *
+ * This screen contains the following actions:
+ *
+ * - Add New Vendor: Navigates to the vendor creation page when the "Add New Vendor" button is clicked.
+ * - View Vendor Details: The user can click on a row to navigate to the detailed view of that vendor.
+ */
+
+@Component({
+  selector: 'app-listing',
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: update selector based on component name

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-creation',
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: update selector

**Review by athul-sanjeevan — APPROVED** [review]

**Review by kavyamolr — APPROVED** [review]

**Comment by NevinJoseph22** [comment]
Merging the UI: Vendor listing HTML and Angular code with LLD

**Pull request merged**